### ISSUE TITLE
confuse on CASIA_train_test.prototxt training loss

### ISSUE THREAD

**Issue by vzvzx** [happynear/FaceVerification#2]
I use the CASIA_train_test.prototxt with my dataset(2000person and 50 image for each total 10W+).
I want to training deepid network using CASIA_train_test.prototxt. Only change the "ip1" layer output num from 10575 to 2000.
but it cannot make the soft loss reduce, and the test accuracy is very small.

In the right case, what's the accuracy will be achieved?
and what about the loss?

my alignment data like this.
![aaron_eckhart_001](https://cloud.githubusercontent.com/assets/2182214/8510646/233476ca-2326-11e5-9998-6e3ab19516e0.jpg)
![aaron_eckhart_002](https://cloud.githubusercontent.com/assets/2182214/8510648/2a30f5ac-2326-11e5-9873-a1e2c3c02dda.jpg)

**Comment by happynear** [comment]
This is a very difficult problem, which has confused deep learning researchers for years. All I can do is to give you some suggestions.
1. Reduce the original lr, e.g. set lr=0.001 at first.
2. Have you used the 'msra' weight filler? In my experience, it works well for initiate the weights.
3. Use smaller network first.

**Comment by vzvzx** [comment]
1.yes, I try set the lr from 0.1 to 0.0001, but it do not work fine.
2.yes I used the 'msra' weight filler and try the 'xavier' weight filler too
3.in small dataset (20 persons) it can achieve the accuracy to 80%. but not work on whole dataset
4.I try the mnist network can achieve 50% accuracy, but not good enough.

can you share a trained model to me, then I can see some detail, or I think I can do some finetune  work.

**Comment by happynear** [comment]
Why not train a model using CASIA-Webface by yourself?

**Comment by vzvzx** [comment]
Yes, I'm training the model. But it's very useful to have some information from your success models.
In new process I have made the loss reduce. :) and the test accuracy have achieved 70%(whole dataset).

In your success model, what test accuracy will be achieved?

![untitled](https://cloud.githubusercontent.com/assets/2182214/8514767/e32ff704-23c8-11e5-8e82-639a7c15d5b2.png)

**Comment by happynear** [comment]
I did not set a test set. I used all the CASIA-webface as training set, and copy them to be the test set.

The final accuracy on training set is about 89.5%~91.5%. As I remembered, after training with lr=0.01, the accuracy is about 80%.

**Comment by vzvzx** [comment]
y, this is very useful information.
with lr=0.01 my accuracy is 82.81% in training datasets, but 66.08% in test datasets.

I that mean I can do Verification based on this models.
![untitled](https://cloud.githubusercontent.com/assets/2182214/8521281/c2025e9a-2414-11e5-8598-ebc300cb5c88.png)

**Comment by vzvzx** [comment]
y, this is very useful information.
with lr=0.01 my accuracy is 82.81% in training datasets, but 66.08% in test datasets.

I that mean I can do Verification based on this models?
![untitled](https://cloud.githubusercontent.com/assets/2182214/8521311/fbb72aa8-2414-11e5-904b-627afdf71359.png)

**Comment by vzvzx** [comment]
y, this is very useful information.
with lr=0.01 my accuracy is 82.81% in training datasets, but 66.08% in test datasets.

Is  that mean I can do Verification based on this models?
![untitled](https://cloud.githubusercontent.com/assets/2182214/8521321/1a3089a2-2415-11e5-95bc-c4a36232ad1f.png)

**Comment by vzvzx** [comment]
thanks, this is very useful information.
with lr=0.01 my accuracy is 82.81% in training datasets, but 66.08% in test datasets.

Is that mean I can do Verification based on this models?
![untitled](https://cloud.githubusercontent.com/assets/2182214/8521410/e2814edc-2415-11e5-88da-514789aedec5.png)

**Comment by happynear** [comment]
You should at least continue training with lr=0.001 for about 150,000 iteration.

**Comment by vzvzx** [comment]
after lr=0.001 my training accuracy achieve ~100%, and test accuracy achieve ~77%
but the verification on lfw with this model, can only achieve ~60% using VerificationDemo.m.

this very strange, there is something wrong with my job.

here is some diff from your process.
by your alignment code, the output data will link this .
![aaron_eckhart_003_1](https://cloud.githubusercontent.com/assets/2182214/8540227/27cdecb6-24b6-11e5-8100-7de9ab6246ca.jpg)
but my crop the face to full the image, is this right?
![aaron_eckhart_003](https://cloud.githubusercontent.com/assets/2182214/8540233/3611a736-24b6-11e5-89b1-bd9fbb4ad396.jpg)

**Comment by happynear** [comment]
I just follow the CASIA-Webface for setting the parameter. You should alignment the face as what you have done with your training data.

**Comment by georgePashchenko** [comment]
Hello! 
I also used CASIA_train_test.prototxt and mnist_siamese_solver.prototxt to train on  normalized CASIA WebFace dataset.  Strange things that soft loss reduces (very slow) , and the test accuracy is not. For example for 10 persons test accuracy is 28%. I changed lr from 0.1 to 0.0001 but result is the same. Any suggestions?
